processing/src/main/java/org/apache/druid/frame/channel/ComposingWritableFrameChannel.java

```diff
@@ -105,6 +100,16 @@ public void write(FrameWithPartition frameWithPartition) throws IOException
     }
   }
 
+  private void convertChannelSuppliersToReadOnly(int index)
```

Comment:
Yes that is part of a refactor that can be done as a follow up.

Reply:
Regarding the common interface, I assume that there can be a lot more common code paths. OutputChannel is essentially a PartitionedOutputChannel where the partition is fixed. This would require refactoring of ReadableFrameChannel as well.

We can do without storing the readOnly() version in the `ComposingOutputChannelFactory`, though I think we can leave it as is, since its immutable and conveys the intent that we on fetching the readable channel, we don't want other references to be stored (if they are). WDYT?